exonum/src/proto/tests.rs

```diff
@@ -326,3 +328,36 @@ fn test_convert_any_message() {
     assert_eq!(artifact_id.runtime_id, 0);
     assert_eq!(artifact_id.name, "exonum-cryptocurrency-advanced/0.11.0");
 }
+
+#[test]
+fn serialize_map_proof() {
+    let db = TemporaryDB::default();
+    let storage = db.fork();
+
+    let mut table = ProofMapIndex::new("index", &storage);
+
+    let proof = table.get_proof(0);
+    assert_deserialized_proof(proof);
+
+    for i in 0..10 {
+        table.put(&i, i);
+    }
+
+    let proof = table.get_proof(5);
+    assert_deserialized_proof(proof);
+
+    let proof = table.get_multiproof(5..15);
+    assert_deserialized_proof(proof);
+}
+
+fn assert_deserialized_proof<K, V>(proof: MapProof<K, V>)
+where
+    K: BinaryKey + ObjectHash + fmt::Debug,
+    V: BinaryValue + ObjectHash + fmt::Debug,
+    MapProof<K, V>: ProtobufConvert + PartialEq,
+{
+    let pb = proof.to_pb();
+    let deserialized: MapProof<K, V> = MapProof::from_pb(pb).unwrap();
+    assert_eq!(proof, deserialized);
+    deserialized.check().expect("proof is not valid");
```

Comment:
Maybe add a check that root hashes from original and deserialized proofs are equal?